pkg/repositories/api_key_repository.go

```diff
@@ -10,11 +10,13 @@ import (
 )
 
 type ApiKeyRepository interface {
-	Create(ctx context.Context, tx db.Transaction, apiKey persistence.ApiKey) (persistence.ApiKey, error)
+	Create(ctx context.Context, apiKey persistence.ApiKey) (persistence.ApiKey, error)
 	Get(ctx context.Context, id uuid.UUID) (persistence.ApiKey, error)
 	GetForKey(ctx context.Context, apiKey uuid.UUID) (persistence.ApiKey, error)
-	GetForUser(ctx context.Context, tx db.Transaction, user uuid.UUID) ([]uuid.UUID, error)
-	Delete(ctx context.Context, tx db.Transaction, ids []uuid.UUID) error
+	GetForUser(ctx context.Context, user uuid.UUID) ([]uuid.UUID, error)
+	GetForUserTx(ctx context.Context, tx db.Transaction, user uuid.UUID) ([]uuid.UUID, error)
```

Comment:
On the other hand we can now delete the keys and **not** the user so we need a version of this method that does not take a transaction.